editor/file_system/editor_file_system.cpp

```diff
@@ -1450,83 +1842,65 @@ void EditorFileSystem::_scan_fs_changes(EditorFileSystemDirectory *p_dir, ScanPr
 
 				int idx = p_dir->find_dir_index(f);
 				if (idx == -1) {
-					String dir_path = cd.path_join(f);
+					const String dir_path = cd.path_join(f);
 					if (_should_skip_directory(dir_path)) {
 						continue;
 					}
 
-					ScannedDirectory sd;
-					sd.name = f;
-					sd.full_path = dir_path;
-
 					EditorFileSystemDirectory *efd = memnew(EditorFileSystemDirectory);
 					efd->parent = p_dir;
 					efd->name = f;
 
-					Ref<DirAccess> d = DirAccess::create(DirAccess::ACCESS_RESOURCES);
-					d->change_dir(dir_path);
-					int nb_files_dir = _scan_new_dir(&sd, d);
-					p_progress.hi += nb_files_dir;
-					diff_nb_files += nb_files_dir;
-					_process_file_system(&sd, efd, p_progress, nullptr);
-
-					ItemAction ia;
-					ia.action = ItemAction::ACTION_DIR_ADD;
-					ia.dir = p_dir;
-					ia.file = f;
-					ia.new_dir = efd;
-					scan_actions.push_back(ia);
+					for (idx = 0; idx < p_dir->subdirs.size(); idx++) {
+						if (efd->name.filenocasecmp_to(p_dir->subdirs[idx]->name) < 0) {
+							break;
+						}
+					}
+					if (idx == p_dir->subdirs.size()) {
+						p_dir->subdirs.push_back(efd);
+					} else {
+						p_dir->subdirs.insert(idx, efd);
+					}
+
+					if (first_scan) {
+						Vector<String> dirs = dir_path.substr(6, dir_path.length() - 6).split("/");
```

Comment:
```suggestion
						Vector<String> dirs = dir_path.substr(6).split("/");
```
Redundant, that's the full remaining string